drivers/sx127x/sx127x.c

```diff
@@ -0,0 +1,419 @@
+
+    sx127x_set_tx_power(dev, SX127X_RADIO_TX_POWER);
+}
+
+uint32_t sx127x_random(sx127x_t *dev)
+{
+    uint32_t rnd = 0;
+
+    sx127x_set_modem(dev, SX127X_MODEM_LORA); /* Set LoRa modem ON */
+
+    /* Disable LoRa modem interrupts */
+    sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGSMASK, SX127X_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                     SX127X_RF_LORA_IRQFLAGS_RXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                     SX127X_RF_LORA_IRQFLAGS_VALIDHEADER |
+                     SX127X_RF_LORA_IRQFLAGS_TXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_CADDONE |
+                     SX127X_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                     SX127X_RF_LORA_IRQFLAGS_CADDETECTED);
+
+    /* Set radio in continuous reception */
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_RECEIVER);
+
+    for (unsigned i = 0; i < 32; i++) {
+        xtimer_usleep(1000); /* wait for the chaos */
+
+        /* Non-filtered RSSI value reading. Only takes the LSB value */
+        rnd |= ((uint32_t) sx127x_reg_read(dev, SX127X_REG_LR_RSSIWIDEBAND) & 0x01) << i;
+    }
+
+    sx127x_set_sleep(dev);
+
+    return rnd;
+}
+
+/**
+ * IRQ handlers
+ */
+void sx127x_isr(netdev_t *dev)
+{
+    if (dev->event_callback) {
+        dev->event_callback(dev, NETDEV_EVENT_ISR);
+    }
+}
+
+static void sx127x_on_dio_isr(sx127x_t *dev, sx127x_flags_t flag)
+{
+    dev->irq |= flag;
+    sx127x_isr((netdev_t *)dev);
+}
+
+static void sx127x_on_dio0_isr(void *arg)
+{
+    sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO0);
+}
+
+static void sx127x_on_dio1_isr(void *arg)
+{
+    sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO1);
+}
+
+static void sx127x_on_dio2_isr(void *arg)
+{
+    sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO2);
+}
+
+static void sx127x_on_dio3_isr(void *arg)
+{
+    sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO3);
+}
+
+/* Internal event handlers */
+void sx127x_on_dio0(void *arg)
+{
+    sx127x_t *dev = (sx127x_t *) arg;
+    netdev_t *netdev = (netdev_t*) &dev->netdev;
+
+    switch (dev->settings.state) {
+        case SX127X_RF_RX_RUNNING:
+            netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
+            break;
+        case SX127X_RF_TX_RUNNING:
+            xtimer_remove(&dev->_internal.tx_timeout_timer);
+            switch (dev->settings.modem) {
+                case SX127X_MODEM_LORA:
+                    /* Clear IRQ */
+                    sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGS,
+                                     SX127X_RF_LORA_IRQFLAGS_TXDONE);
+                /* Intentional fall-through */
+                case SX127X_MODEM_FSK:
+                default:
+                    sx127x_set_state(dev, SX127X_RF_IDLE);
+                    netdev->event_callback(netdev, NETDEV_EVENT_TX_COMPLETE);
+                    break;
+            }
+            break;
+        default:
+            puts("sx127x_on_dio0: Unknown state");
```

Comment:
@aabadie I encountered a potential bug : 
I sent 7 bytes "123456\0" using tests/driver_sx127x and the send command
If I keep the touch enter pressed (yes I'm a monster) It works so far.
Then I re-press the enter key and I get
sx127x_on_dio0: Unknown state
Note: data are emitted ! But there is no more Transmission completed message anymore (no tx done interrupt received ? Weird...)
I'm now wondering : If this case happens in reception. No more RX done interrupt no more message received -> we're stuck.
Note : if I get sx127x_on_dio0: Unknown state, then call listen command and try to re-sent data after that. Rx done interrupt comes back.
There might be something to do...
I'm now wondering :